gammapy/spectrum/dataset.py

```diff
@@ -282,3 +280,63 @@ def energy_range(self):
         e_lo = self.counts_on.energy.lo[self.mask]
         e_hi = self.counts_on.energy.hi[self.mask]
         return u.Quantity([e_lo.min(), e_hi.max()])
+
+    def _as_counts_spectrum(self, data):
+        energy = self.counts_on.energy
+        return CountsSpectrum(data=data, energy_lo=energy.lo, energy_hi=energy.hi)
+
+    def excess(self):
+        """Excess (counts_on - alpha * counts_off)"""
+        excess = self.counts_on.data.data - self.alpha * self.counts_off.data.data
+        return self._as_counts_spectrum(excess)
+
+    def residuals(self):
+        """Residuals (npred - excess). """
+        residuals = self.npred().data.data - self.excess().data.data
+        return self._as_counts_spectrum(residuals)
+
+    def peek(self):
+        """Plot counts and residuals in two panels.
+
+        Calls ``plot_counts`` and ``plot_residuals``.
+        """
+        from matplotlib.gridspec import GridSpec
+        import matplotlib.pyplot as plt
+
+        gs = GridSpec(7, 1)
+
+        ax_spectrum = plt.subplot(gs[:5, :])
+        self.plot_counts(ax=ax_spectrum)
+
+        ax_spectrum.set_xticks([])
+
+        ax_residuals = plt.subplot(gs[5:, :])
+        self.plot_residuals(ax=ax_residuals)
+        return ax_spectrum, ax_residuals
+
+    def plot_counts(self, ax):
+        """Plot predicted and detected counts."""
+        self.npred().plot(ax=ax, label="mu_src")
+        self.excess().plot(ax=ax, label="Excess", fmt=".", energy_unit="TeV")
```

Comment:
Why forcing the units to be TeV? That might be optional? Or simply use the units of the counts?

Reply:
Done